src/state-react/useSubscription.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import { State, IMergedState } from '../state';
+
+function invokeAsPromise(delegate) {
+  try {
+    return Promise.resolve(delegate());
+  } catch (e) {
+    return Promise.reject(e);
+  }
+}
```

Comment:
Make sure that synchronous and asynchronous errors are handled and turn into rejected promises.

Normally this is all handled by just doing try/catch with async/await but, hooks :)

Reply:
`return Promise.resolve().then(delegate)`

Reply:
you won't need this utility anymore